### PR TITLE
April Leet-code questions

### PR DESCRIPTION
**Pull request by Rahul171201**
#54 Added April1 and April22 leet-code challenge solutions!

### CODE REVIEW

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
Please write code for the function headers in Leetcode. i.e.
```
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
    }
};
```
and
```
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
    }
};
```

**Comment by Rahul171201** [comment]
ok, working on it

File: Leetcoding-April1.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    int singleNumber(vector<int>& nums) {
+        
+        int x;
+        while(cin>>x){
+            nums.push_back(x);
+        }
+        	int B[1000001];
+	for(int i=0;i<=1000000;i++)
+	B[i]=0;
+	for(int i=0;i<nums.length;i++)
```
Comment (priyanshisharma) [inline Leetcoding-April1.cpp:12]:
```suggestion
	for(int i=0;i<nums.size();i++)

```

File: Leetcoding-April1.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	for(int i=0;i<=1000000;i++)
+	{
+		if(B[i]==1)
+		{
+			cout<<i;
+			break;
+		}
+	}
+
```
Comment (priyanshisharma) [inline Leetcoding-April1.cpp:25]: No Return statement

File: Leetcoding-April1.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class Solution {
+public:
+    int singleNumber(vector<int>& nums) {
+        
+        int x;
+        while(cin>>x){
```
Comment (priyanshisharma) [inline Leetcoding-April1.cpp:6]: Input is in `nums`. No console input is required.

File: Leetcoding-April22.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Solution {
+public:
+    int subarraySum(vector<int>& nums, int k) {
+        
+        int x;
+        while(cin>>x)
+            nums.push_back(x);
+	
+	int count=0;
+	 for (int i = 0; i < nums.length; i++) {
```
Comment (priyanshisharma) [inline Leetcoding-April22.cpp:10]:
```suggestion
	 for (int i = 0; i < nums.size(); i++) {   

```

File: Leetcoding-April22.cpp [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    int subarraySum(vector<int>& nums, int k) {
+        
+        int x;
+        while(cin>>x)
+            nums.push_back(x);
+	
+	int count=0;
+	 for (int i = 0; i < nums.length; i++) {   
+            int sum = 0;
+            for (int j = i; j < n; j++) {
```
Comment (priyanshisharma) [inline Leetcoding-April22.cpp:12]:
```suggestion
            for (int j = i; j < nums.size(); j++) {

```

File: Leetcoding-April22.cpp [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            	
+                sum += A[j];
```
Comment (priyanshisharma) [inline Leetcoding-April22.cpp:14]:
```suggestion
                sum += nums[j];

```

File: Leetcoding-April22.cpp [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+                if (sum == k)                   
+                   count++;
+            }
+        }
+        cout<<count;
```
Comment (priyanshisharma) [inline Leetcoding-April22.cpp:19]:
```suggestion
       return count;

```

File: Leetcoding-April22.cpp [hunk 8]
```diff
@@ -0,0 +1,23 @@
+class Solution {
+public:
+    int subarraySum(vector<int>& nums, int k) {
+        
+        int x;
+        while(cin>>x)
```
Comment (priyanshisharma) [inline Leetcoding-April22.cpp:6]: No console input is required

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
Please write your code according to the function definitions. Feel free to read the question on leetcode.

**Comment by Rahul171201** [comment]
I am really sorry @priyanshisharma , I should have had a good look at the question. just sending the final solutions.
Thankyou

**Comment by Rahul171201** [comment]
Final updated solutions. thankyou for help @priyanshisharma

**Review by priyanshisharma — CHANGES_REQUESTED** [review]
Its alright 🙌 
Your April1 solution fails for input `[-1]`, please debug the same.

**Comment by Rahul171201** [comment]
ok sure

**Review by priyanshisharma — APPROVED** [review]

**Pull request merged**